components/Highlight.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace app\components;
+
+use yii\base\Widget;
+use yii\helpers\Html;
+
+/**
+ * Highlight widget renders the content between it's begin end tags.
+ *
+ * Usage example:
+ *
+ * ```php
+ * <body>
+ *     <?php Highlight::begin(); ?>
+ *         <div class="nav-bar">
+ *         </div>
+ *         <div class="content">
+ *         </div>
+ *     <?php Highlight::end(); ?>
+ * </body>
+ * ```
+ */
+class Highlight extends Widget
+{
+    // what language to highlight in
+    public $language = 'php';
+
+    // if $capture is set to true
+    // then $captured will be set to
+    // the highlighted result and not returned
+    public $capture = false;
+    public $captured = null;
+
+    protected $highlighter = null;
+
+    /**
+     * Starts capturing an output to be highlighted.
+     */
+    public function init()
+    {
+        parent::init();
+        ob_start();
+    }
+
+    protected function getHighlighter()
+    {
+        if(!isset($this->highlighter))
```

Comment:
Could be simplified:

```php
```php
if (!isset($this->highlighter)) {
    $this->highlighter = new \Highlight\Highlighter();
}
return $this->highlighter;
```